components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -374,19 +381,23 @@ public String importIDP(InputStream fileInputStream, Attachment fileDetail) {
      * @param identityProviderId Resource ID of the Identity Provider to be updated.
      * @param fileInputStream    File to be imported as an input stream.
      * @param fileDetail         File details.
-     * @return Unique identifier of the updated identity provider.
      */
-    public String updateIDPFromFile(String identityProviderId, InputStream fileInputStream, Attachment fileDetail) {
+    public void updateIDPFromFile(String identityProviderId, InputStream fileInputStream, Attachment fileDetail) {
```

Comment:
why did we have to change the return type?

Reply:
Earlier as an optional header we passed the api path with the resourceId in the response header so that a user can directly use that path and call the GET method to get that idp's details [[1](https://github.com/wso2/identity-api-server/pull/452/files#diff-a3f5e5ae7f7ca102f8e752a15cffe223800191dcd4b0637a608e5f56dd5db016L265)]. This is normally used when a new resource is created with a POST method (in most update apis we do not send this). Here we cannot create the location that way for resident api, since we do not have a GET method to get resident api details.

[1] https://github.com/wso2/identity-api-server/pull/452/files#diff-a3f5e5ae7f7ca102f8e752a15cffe223800191dcd4b0637a608e5f56dd5db016L265